File: server/src/main/java/by/brawl/entity/Account.java
package by.brawl.entity;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import javax.persistence.*;
import java.util.*;

/**
 * Default class description.
 *
 * @author P.Sinitsky.
 *         Created on 25.03.2017.
 */
@Entity
public class Account extends IdEntity implements UserDetails {

    private String email;
    private String password;

    @OneToMany(mappedBy = "owner")
    private Set<Hero> heroes = new LinkedHashSet<>();
    @OneToMany(mappedBy = "owner")
    private Set<Squad> squads = new LinkedHashSet<>();

    @ManyToMany
    @JoinTable(name = "account_spell",
            joinColumns = @JoinColumn(name = "account_id"),
            inverseJoinColumns= @JoinColumn(name = "spell_id"))
    private Set<Spell> availableSpells = new LinkedHashSet<>();

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return null;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return email;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }

    public Set<Spell> getAvailableSpells() {
        return availableSpells;
    }

    public Set<Squad> getSquads() {
        return squads;
    }

    public Set<Hero> getHeroes() {
        return heroes;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Account)) return false;
        Account account = (Account) o;
        return Objects.equals(getId(), account.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), email, password, heroes, squads, availableSpells);
    }

    @Override
    public String toString() {
        return "Account{" +
                "email='" + email + '\'' +
                '}';
    }
}

File: server/src/main/java/by/brawl/entity/Account.kt
package by.brawl.entity

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.util.*
import javax.persistence.*

@Entity
class Account : IdEntity(), UserDetails {

    private lateinit var email: String
    private lateinit var password: String

    @OneToMany(mappedBy = "owner")
    val heroes: Set<Hero> = LinkedHashSet()
    @OneToMany(mappedBy = "owner")
    val squads: Set<Squad> = LinkedHashSet()

    @ManyToMany
    @JoinTable(name = "account_spell",
            joinColumns = arrayOf(JoinColumn(name = "account_id")),
            inverseJoinColumns = arrayOf(JoinColumn(name = "spell_id")))
    val availableSpells: Set<Spell> = LinkedHashSet()

    override fun getAuthorities(): Collection<GrantedAuthority> {
        return listOf()
    }

    override fun getPassword(): String {
        return password
    }

    override fun getUsername(): String {
        return email
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }

    override fun toString(): String {
        return "Account{" +
                "email='" + email + '\'' +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Account) return false

        if (email != other.email) return false

        return true
    }

    override fun hashCode(): Int {
        var result = email.hashCode()
        result = 31 * result + password.hashCode()
        result = 31 * result + heroes.hashCode()
        result = 31 * result + squads.hashCode()
        result = 31 * result + availableSpells.hashCode()
        return result
    }

}